### PR TITLE
Theming whitelabel.py script

### PR DESCRIPTION
**Pull request by rnr**
# Added the theming `whitelabel.py` script

This script can override Assets (images, colors, app Icon) and change project settings (app bundle ID and development team ID). Please follow [this documentation](https://github.com/touchapp/openedx-app-ios/blob/config-whitelabel-script/Documentation/Theming_implementation.md) to install Python dependencies if needed and to read how to work/configure.

### 1) Allocated colors for more flexibility:

- `loginNavigationText`
- `loginBackground`
- `accentButtonColor`
- `SplashBackground`

## Custom edX theme example

1. Clone `https://github.com/edx/edx-mobile-config` into the OpenEdX root folder and switch to the `anton/openedX-theming` branch.

2. From the OpenEdX root folder, run the script as:
```bash
python config_script/whitelabel.py --config-file=edx-mobile-config/openEdXAssets/whitelabel.yaml -v
```
3. Install on your simulator or device

All possible config parameters are described in [this documentation](https://github.com/touchapp/openedx-app-ios/blob/config-whitelabel-script/Documentation/Theming_implementation.md)

### CODE REVIEW

**Comment by miankhalid** [comment]
@shafqat-muneer will be reviewing this PR from Axinite's side.

**Comment by shafqat-muneer** [comment]
PR review is in progress from my side.

**Comment by shafqat-muneer** [comment]
After running this command on terminal:
```
python3 config_script/whitelabel.py --config-file=path/to/configfile/whitelabel.yaml -v
```

Got this error:
```
File "/path/to/config_script/whitelabel.py", line 6, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
```

Resolved the issue by installing `pyyaml` by following command:
```
pip3 install pyyaml
```

**Suggestion:** I think we should document every dependency in `Theming_implementation.md` file so that it can easily be configured on new system independently.

**Comment by shafqat-muneer** [comment]
We utilize distinct bundle identifiers for our Development (Dev), Staging (Stage), and Production (Prod) environments. Is it possible to modify the bundle identifiers for all environments simultaneously, or do we need to execute the script multiple times to update each environment's bundle identifier individually?

**Review by shafqat-muneer — CHANGES_REQUESTED** [review]

**Comment by shafqat-muneer** [comment]
Review is in progress.

**Comment by shafqat-muneer** [comment]
@rnr Getting `SyntaxWarning` after running command:
```
python3 config_script/whitelabel.py --config-file=/path/to/whitelabel.yaml
```

Error Details:
```
/path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
  return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
/path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
  return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
/path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\s'
  return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
```

**Comment by rnr** [comment]
> @rnr Getting `SyntaxWarning` after running command:
> 
> ```
> python3 config_script/whitelabel.py --config-file=/path/to/whitelabel.yaml
> ```
> 
> Error Details:
> 
> ```
> /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
>   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
>   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\s'
>   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> ```

@shafqat-muneer  Could you show what is content of whitelabel.yaml file for this error please

**Comment by shafqat-muneer** [comment]
> > @rnr Getting `SyntaxWarning` after running command:
> > ```
> > python3 config_script/whitelabel.py --config-file=/path/to/whitelabel.yaml
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Error Details:
> > ```
> > /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
> >   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> > /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
> >   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> > /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\s'
> >   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> > ```
> 
> @shafqat-muneer Could you show what is content of whitelabel.yaml file for this error please

@rnr I tried it with just `import_dir` content. `whitelabel.yaml` looks like this:
```
import_dir: '/Users/shafqatmuneer/msmq/Projects/2U/_Daily/20231226/Images' # folder where importing images are placed
```

I also checked with same sample file content in config repo, it's also giving error. With `whitelabel.yaml` file in config repo, error is this:
```
/Users/shafqatmuneer/msmq/Projects/2U/CodeBase/openedx-app-ios/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
  return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
/Users/shafqatmuneer/msmq/Projects/2U/CodeBase/openedx-app-ios/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
  return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
/Users/shafqatmuneer/msmq/Projects/2U/CodeBase/openedx-app-ios/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\s'
  return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
```

File: Documentation/Theming_implementation.md
Comment (rnr) [reply Documentation/Theming_implementation.md]: added to documentation all dependencies which are not default python's libs

File: Documentation/Theming_implementation.md
Comment (rnr) [reply Documentation/Theming_implementation.md]: fixed

File: config_script/whitelabel.py
Comment (rnr) [reply config_script/whitelabel.py]: fixed

**Comment by rnr** [comment]
> > > @rnr Getting `SyntaxWarning` after running command:
> > > ```
> > > python3 config_script/whitelabel.py --config-file=/path/to/whitelabel.yaml
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Error Details:
> > > ```
> > > /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
> > >   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> > > /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
> > >   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> > > /path/to/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\s'
> > >   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> > > ```
> > 
> > 
> > @shafqat-muneer Could you show what is content of whitelabel.yaml file for this error please
> 
> @rnr I tried it with just `import_dir` content. `whitelabel.yaml` looks like this:
> 
> ```
> import_dir: '/Users/shafqatmuneer/msmq/Projects/2U/_Daily/20231226/Images' # folder where importing images are placed
> ```
> 
> I also checked with same sample file content in config repo, it's also giving error. With `whitelabel.yaml` file in config repo, error is this:
> 
> ```
> /Users/shafqatmuneer/msmq/Projects/2U/CodeBase/openedx-app-ios/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
>   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> /Users/shafqatmuneer/msmq/Projects/2U/CodeBase/openedx-app-ios/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\*'
>   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> /Users/shafqatmuneer/msmq/Projects/2U/CodeBase/openedx-app-ios/config_script/whitelabel.py:317: SyntaxWarning: invalid escape sequence '\s'
>   return f"/\* {build_config} \*/ = {{\n\t\t\tisa = XCBuildConfiguration;\n\t\t\tbaseConfigurationReference = [\s|\S]*\t\t\tname = {build_config};"
> ```

@shafqat-muneer added extra escaping to regex inside python script. should be fine now. please check

**Review by shafqat-muneer — APPROVED** [review]
LGTM
- Script is working
- Documentation related improvements are done
- Dependencies are listed
- Sample `whitelabel.yaml` file added

Screenshot 2023-12-27 at 9 08 04 PM

**Comment by volodymyr-chekyrta** [comment]
Hello everyone!

I was wondering if we are waiting for @eyatsenkoperpetio review before merging this PR, or if we can go ahead and merge it now?

**Review by eyatsenkoperpetio — APPROVED** [review]

**Comment by rnr** [comment]
> Hello everyone!
> 
> I was wondering if we are waiting for @eyatsenkoperpetio review before merging this PR, or if we can go ahead and merge it now?

Hello @volodymyr-chekyrta 
I think we are ready to merge since @eyatsenkoperpetio just approved 
Thank you

**Pull request merged**
